staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -1034,6 +1116,29 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 	return ret, nil
 }
 
+func writeStorageVersion(ctx context.Context, crdInfo *crdInfo) {
+	// wait for previous handlers to complete
+	time.Sleep(2 * time.Minute) // make this use an old waitgroup or some such
```

Comment:
Yeah would need some foolproof way of knowing when the previous handlers have torn down completely.